assets/js/components/Collapsible.js

```diff
@@ -0,0 +1,182 @@
+'use strict';
+
+const utils = require('../libs/elife-utils')();
+
+module.exports = class Collapsible {
+
+  constructor($elm, _window = window, doc = document) {
+
+    this.$elm = $elm;
+    this.window = _window;
+    this.doc = doc;
+
+    this.viewportWidthMedium = 729;
+    this.viewportWidthLarge = 899;
+    this.maxVisibleCollapsedElementsOnSmallViewport = 2;
+    this.maxVisibleCollapsedElementsOnMediumViewport = 6;
+    this.listElements = Array.prototype.slice.call($elm.children);
+
+    if (this.viewportNoWiderThan(this.viewportWidthMedium) &&
+      this.listElements.length > this.maxVisibleCollapsedElementsOnSmallViewport) {
+      this.$maxVisibleCollapsedElements = this.maxVisibleCollapsedElementsOnSmallViewport;
+      this.initialise();
+    } else if (this.viewportNoWiderThan(this.viewportWidthLarge) &&
+      this.listElements.length > this.maxVisibleCollapsedElementsOnMediumViewport) {
+      this.$maxVisibleCollapsedElements = this.maxVisibleCollapsedElementsOnMediumViewport;
+      this.initialise();
+    }
+
+    this.window.addEventListener('resize', this.initialiseOnResize.bind(this));
+  }
+
+  initialise() {
+    this.$toggle = this.createToggle();
```

Comment:
`initialise()` can be run multiple times and some repeated steps should be avoided. If we have created a toggle element before then we have no need to recreate it we can just have a way of resetting toggle. 

What is the expected behaviour on resize? If we have expanded the definition list prior to resize do we expect for it to remain expanded on resize? cc @chris-huggins